usr/share/rear/build/default/500_patch_sshd_config.sh

```diff
@@ -1,33 +1,102 @@
 #
-# adapt some ssh configs
+# Adapt some SSH configs and as needed regenerate SSH host key:
+
+# There is nothing to do when there are no SSH binaries on the original system:
+has_binary ssh || has_binary sshd || return
+
+# Do nothing when not any SSH file should be copied into the recovery system:
+is_false "$SSH_FILES" && return
```

Comment:
Is this intentional? SSH login could still be allowed without copying any ssh configuration to the rescue system by just creating a host key. If the intention is to block SSH login when SSH_FILES is false, it might be a good idea to show a warning if SSH_ROOT_PASSWORD is set (as this will not work).

Reply:
At least as a first attempt how SSH_FILES is meant to work
SSH_FILES="no"
should mean
"I do not want that the ssh scripts in ReaR
 do anything - if I need SSH stuff I do it myself
 via COPY_AS_IS and REQUIRED_PROGS"

I will derscibe the current meaning of SSH_FILES
in default.conf and add that via another commit here.

Until the ReaR 2.3 release it can be changed as needed
because the current ReaR master code is under development.

Reply:
And I will show a LogPrintError info
if SSH_ROOT_PASSWORD is set
together with SSH_FILES="no"